yarn-project/end-to-end/src/e2e_token_contract.test.ts

```diff
@@ -496,6 +496,31 @@ describe('e2e_token_contract', () => {
           );
         });
 
+        it('transfer on behalf of other after deleting approval', async () => {
+          const balance0 = await asset.methods.balance_of_private(accounts[0].address).view();
+          const amount = balance0 / 2n;
+          const nonce = Fr.random();
+          // expect(amount).toBeGreaterThan(0n);
+
+          // docs:start:remove_private_auth_witness
+          // create auth witness
+          const action = asset
+            .withWallet(wallets[1])
+            .methods.transfer(accounts[0].address, accounts[1].address, amount, nonce);
+          const messageHash = await computeAuthWitMessageHash(accounts[1].address, action.request());
+          const witness = await wallets[0].createAuthWitness(messageHash);
+          await wallets[1].addAuthWitness(witness);
+
+          // wallets[0] want to undo their authorization action. So they tell wallets[1] to do so:
+          await wallets[1].cancelPrivateAuthWitness(messageHash);
```

Comment:
Not sure this test requires two wallets to demonstrate the functionality

Reply:
no, but it makes it explicit that 2nd wallet has to do it and not the first one (external community sometimes looks at our e2e tests and this makes things clear)